### PR TITLE
Component options section

### PR DESCRIPTION
**Pull request by tivac**
Not merging this one immediately because I'm worried there are subtleties that I'm missing and it's late.

### CODE REVIEW

**Comment by tivac** [comment]
Will verify against the code or the tests in the morning hopefully.

**Comment by isiahmeadows** [comment]
@tivac I thought it was reduced to `vnode.props` and `vnode.children`, with `m(Component, props?, ...children)`.

File: docs/v1.x-migration.md [hunk 1]
```diff
@@ -120,6 +121,42 @@ m.mount(document.body, {
 });
 ```
 
+## Component arguments
+
+Passing arguments to a component hasn't changed, but how you access them within the component has changed.
```
Comment (StephanHoyer) [inline docs/v1.x-migration.md]: that's not quite right ;)
in 0.2.x you could do `m.component(myComp, 'foo')` and got `'foo'` as the second argument on the controller. This seems not working with 1.x, `attrs` always has to be an object.

But I'm not 100% sure. Please dbl-check this.

Comment (lhorie) [inline docs/v1.x-migration.md]: Yeah, now m(Comp, "foo") considers the second argument a child vnode

**Comment by tivac** [comment]
Appreciate the info, mostly started this so I wouldn't forget to document it. Will update later today.

**Comment by tivac** [comment]
Ok did some spelinking, verified my assumptions, clarified some language, and also covered `config: m.route` :arrow_right: `oncreate: m.route.link`.

File: docs/v1.x-migration.md [hunk 2]
```diff
@@ -120,6 +122,42 @@ m.mount(document.body, {
 });
 ```
 
+## Component arguments
+
+Arguments to a component in `v1.x` must be an object, simple values like `String`/`Number`/`Boolean` will be treated as text chidlren. Arguments are accessed within the component by reading them from the `vnode.attrs` object.
```
Comment (lhorie) [inline docs/v1.x-migration.md]: typo: will be treated as text *children*

Comment (tivac) [inline docs/v1.x-migration.md]: :disappointed:

**Comment by tivac** [comment]
@lhorie any further concerns w/ this?

**Comment by lhorie** [comment]
Looks good to me

**Pull request merged**